Engine/TestPlanRun.cs

```diff
@@ -95,6 +95,14 @@ public string TestPlanName
         /// <summary> The thread that started the test plan. Use this to abort the plan thread. </summary>
         public TapThread MainThread { get; }
 
+        internal TestPlanExecuteArgs ExecuteArgs { get; set; }
+
+        /// <summary> Requests that the test plan run gets paused. If UI cannot pause the test plan this will be silently ignored. </summary>
+        public void Pause(TestStepRun stepRun = null)
```

Comment:
This effectively blocks the calling thread until the test plan is resumed, right? Should the doc comment mention this?

Also, I am wondering if this should return a bool indicating whether the test plan paused or not to avoid steps with weird behavior if they expect pausing to be available.